README.md

```diff
@@ -206,6 +206,25 @@ prefix (e.g.`@unpinned_maven` or `@unpinned_<your_maven_install_name>`). For
 example, if your `maven_install` is named `@foo`, `@unpinned_foo` will be
 created.
 
+### Requiring lock file repinning when the list of artifacts changes
+
+It can be easy to forget to update the `maven_install.json` lock file
+when updating artifacts in a `maven_install`. Normally,
+rules_jvm_external will print a warning to the console and continue
+the build when this happens, but by setting the
+`fail_if_repin_required` attribute to `True`, this will be treated as
+a build error, causing the build to fail. When this attribute is set,
+it is possible to update the `maven_install.json` file using:
+
+```shell
+$ REPIN=1 bazel run @unpinned_maven//:pin
```

Comment:
It would be nice if running :pin did not require setting an env var, since running :pin implies that your are trying to re-generate the json file. 

I'm not sure how this would work in the repository rule though since I don't think there is a way to ask the repository context what action is being run.  You could maybe do something where the pin.sh script generates a file that the `_pinned_coursier_fetch_impl` could look for an delete but I'm not sure if it's reliable to share an output from one repository rule to another this way.   @jin do you have any ideas?

Reply:
You're right that's the ideal way for this to work. Unfortunately for us, `bazel run` extends `bazel build`, and this requires the `WORKSPACE` file to be valid and properly configured, so we're caught in a catch-22: we're not correct so we can't run, but if we can't run, we can't be correct. *sigh*

To mitigate this, the default behaviour is to emit a warning rather than fail the build. The user must set `fail_if_repin_required` to `True` to get the build-breaking behaviour. To try and help more, we also print a warning to the console if the build needs to be repinned, and that command is listed.